lib/nmatrix/nmatrix.rb

```diff
@@ -90,6 +90,62 @@ def size(shape)
       shape = [shape,shape] unless shape.is_a?(Array)
       (0...shape.size).inject(1) { |x,i| x * shape[i] }
     end
+
+    # Make N-D coordinate arrays for vectorized evaluations of
+    # N-D scalar/vector fields over N-D grids, given N one-dimensional
+    # coordinate arrays arrs. N > 1.
+    #
+    # call-seq:
+    #     meshgrid(arrs) -> Array of NMatrix
+    #     meshgrid(arrs, options) -> Array of NMatrix
+    #
+    # * *Arguments* :
+    #   - +arrs+ -> Array of N one-dimensional coordinate arrays (Array or NMatrix)
+    #   - +options+ -> Hash with options (:sparse and :ij both Boolean, false by default)
+    # * *Returns* :
+    #   - Array of N N-D NMatrixes
+    # * *Examples* :
+    #     x, y = NMatrix::meshgrid([[1, 2, 3], [4, 5]])
+    #     x.to_a #<= [[1, 2, 3], [1, 2, 3]]
+    #     y.to_a #<= [[4, 4, 4], [5, 5, 5]]
+    #
+    # * *Using* *options* :
+    #
+    #     x, y = NMatrix::meshgrid([[1, 2, 3], [4, 5]], sparse: true)
+    #     x.to_a #<= [[1, 2, 3]]
+    #     y.to_a #<= [[4], [5]]
+    #
+    #     x, y = NMatrix::meshgrid([[1, 2, 3], [4, 5]], ij: true)
+    #     x.to_a #<= [[1, 1], [2, 2], [3, 3]]
+    #     y.to_a #<= [[4, 5], [4, 5], [4, 5]]
+    def meshgrid(arrs, options = {})
+      raise(ArgumentError, 'Expected at least 2 arrays.') if arrs.size < 2
+      mats = arrs.map do |arr|
+        case arr
+        when NMatrix
+          arr
+          raise(ArgumentError, 'Expected one-dimensional arrays.') if arr.effective_dim != 1
+        when Array
+          NMatrix.new([arr.size], arr)
+        else
+          raise(ArgumentError, 'Expected one-dimensional arrays.')
+        end
+      end
```

Comment:
ok then just check with a `..?..:..`?

I think we dont need the checks for 1D matrices so that should not be a problem. Just hold to that for a while until @mohawkjohn takes a look at this. He'll take the final call.